### PR TITLE
Added Search Specific Repo, Refactor Code for getting issues

### PR DESCRIPTION
**Pull request by shakti97**
I got to know that you are using good first issue package which is only supporting some of the repo, so for now the search will work for the supported repos as soon as the supported repos increase this search will be more helpful.

### CODE REVIEW

**Comment by siddhant1** [comment]
image
Is this working correctly?

**Comment by siddhant1** [comment]
Also, I think we should add an always on input bar with a call a search button

**Comment by shakti97** [comment]
> image
> Is this working correctly?

actually this issue occured because of the dependency on good first issue package, if you can see there are specific names provided to the packages and good first issue makes request based on that, so if searched repo name is not same as the package name written in the json file it will unable to find the issues

**Comment by shakti97** [comment]
One fix is migrating from good first issue package to Octokit which will make to prepare your own query and search its issue

**Comment by siddhant1** [comment]
Yup , thats the best option we have rn  , I have opened the issue , will migrate soon!

File: src/components/issues/issueContainer.js [hunk 1]
```diff
@@ -5,36 +5,47 @@ import Issue from "./Issue";
...
   async componentDidUpdate(prevProps) {
     if (!isEqual(prevProps, this.props)) {
       this.setState(current => {
         return { ...current, loading: true };
       });
+      this.fetchIssues();
+    }
+  }
+  fetchIssues = async () => {
+    try {
```
Comment (siddhant1) [inline src/components/issues/issueContainer.js:25]: This is a comment

File: src/components/issues/issueContainer.js [hunk 2]
```diff
@@ -5,36 +5,47 @@ import Issue from "./Issue";
...
   }
   async componentDidUpdate(prevProps) {
     if (!isEqual(prevProps, this.props)) {
       this.setState(current => {
         return { ...current, loading: true };
       });
+      this.fetchIssues();
+    }
+  }
+  fetchIssues = async () => {
```
Comment (siddhant1) [inline src/components/issues/issueContainer.js:24]: This is another comment

File: src/components/issues/issueContainer.js (hunk 1 above)
Comment (siddhant1) [inline src/components/issues/issueContainer.js:25]: THis is a comment

File: src/components/issues/issueContainer.js [hunk 3]
```diff
@@ -5,36 +5,47 @@ import Issue from "./Issue";
 import Spinner from "../reusable/Spinner/spinner";
 
 class IssueContainer extends React.Component {
-  state = { issues: [], loading: false };
+  state = { issues: [], loading: false, error: "" };
```
Comment (siddhant1) [inline src/components/issues/issueContainer.js:8]: Ths is a comment

**Comment by siddhant1** [comment]
sorry dudes , was testing something and unintensionally raised a bunch of comments , closing this for now

**Pull request closed without merging**